src/utils/schema_utils.ts

```diff
@@ -45,34 +55,38 @@ function schemaValidationErrorToValidationErrorItem(schemaValidationErrorObject:
             'dependencies',
         ].includes(schemaValidationErrorObject.keyword)
     ) {
-        return {
-            field: schemaValidationErrorObject.dataPath.replace('.', ''),
-            code: ValidationErrorCodes.IncorrectFormat,
-            reason: schemaValidationErrorObject.message || '',
-        };
+        field = schemaValidationErrorObject.dataPath.replace('.', '');
+        code =  ValidationErrorCodes.IncorrectFormat; 
```

Comment:
small nit, but I actually prefer each `if` block returning its own object. You can accomplish a very similar goal (where all of them share the same reason) by expressing it as such:
```
return {
    field: schemaValidationErrorObject.dataPath.replace('.', ''),
    code: ValidationErrorCodes.IncorrectFormat,
    reason,
}
```

The rationale (both extremely minor)
- No mutation (when you have `let` this makes it easier to do mutation in the future, which can result in hard to reason about code over time)
- All of the logic is localized in the `if` block. Right now, you have to scroll to the bottom of the fn to understand what the return value is. So you need to read the `if` block and scroll down

Reply:
We decided to write it this way because we need a check for whether the `description` field was present. If the field is not present we do not want to populate it in the returned object. If we want to add a return statement in each `if` block we'd need to add the check for `if(schemaValidationErrorObject.description === undefined)` in each `if` block.

Is there a way we can have both the return statement in each `if` block without having a repetitive check for error description in every `if` block?

Reply:
ah yeah, I didn't see that down there. To write it without an `if` statement, you can do the following:
```
//somewhere above
const description = schemaValidationErrorObject.description;

// later
return {
    field: schemaValidationErrorObject.dataPath.replace('.', ''),
    code: ValidationErrorCodes.IncorrectFormat,
    reason,
    ...(description && { description }) 
}
```

`description` can either be `undefined` or a string.

 - If undefined:  `...` operator on `undefined`, it will do nothing - which is what we want (i.e. we don't want `{description: undefined}`)
 - If a string, we are using the `...` operator on an object that looks like `{description: "my error description"}` which is what we want